packages/mco/components/mco-dashboard/data-policy/parsers/subscription-parser.tsx

```diff
@@ -0,0 +1,163 @@
+import * as React from 'react';
+import { APPLICATION_TYPE, DRPC_STATUS } from '@odf/mco/constants';
+import {
+  DisasterRecoveryResourceKind,
+  SubscriptionGroupType,
+  useSubscriptionResourceWatch,
+} from '@odf/mco/hooks';
+import { ACMPlacementModel } from '@odf/mco/models';
+import {
+  ACMManagedClusterKind,
+  ACMPlacementRuleKind,
+  DRClusterAppsMap,
+  DRClusterKind,
+  ProtectedAppsMap,
+} from '@odf/mco/types';
+import {
+  findDRType,
+  getClustersFromDecisions,
+  getProtectedPVCsFromDRPC,
+} from '@odf/mco/utils';
+import { getName, getNamespace } from '@odf/shared/selectors';
+import * as _ from 'lodash-es';
+
+export const useSubscriptionParser: UseSubscriptionParser = (
+  drResources,
+  drLoaded,
+  drLoadError,
+  managedClusters,
+  managedClusterLoaded,
+  managedClusterLoadError
+) => {
+  const [subscriptionResources, loaded, loadError] =
+    useSubscriptionResourceWatch({
+      drResources: {
+        data: drResources,
+        loaded: drLoaded,
+        loadError: drLoadError,
+      },
+    });
+
+  const drClusters: DRClusterKind[] = drResources?.drClusters;
+  const drClusterAppsMap: DRClusterAppsMap = React.useMemo(() => {
+    if (loaded && !loadError) {
+      const drClusterAppsMap: DRClusterAppsMap = drClusters.reduce(
+        (acc, drCluster) => {
+          const clusterName = getName(drCluster);
+          acc[clusterName] = {
+            totalAppCount: 0,
+            protectedApps: [],
+            managedCluster: {},
+          };
+          return acc;
+        },
+        {} as DRClusterAppsMap
+      );
+
+      subscriptionResources.forEach((subscriptionResource) => {
+        const { application, subscriptionGroupInfo } =
+          subscriptionResource || {};
+        const applicationNameSpace = getNamespace(application);
+        const clusterWiseSubscriptionGroups: {
+          [clusterName: string]: SubscriptionGroupType[];
+        } = {};
+
+        subscriptionGroupInfo?.forEach((subscriptionGroup) => {
+          const deploymentClusters: string[] =
+            subscriptionGroup?.placement?.kind === ACMPlacementModel.kind
+              ? getClustersFromDecisions(subscriptionGroup?.placementDecision)
+              : getClustersFromDecisions(
+                  subscriptionGroup?.placement as ACMPlacementRuleKind
+                );
+
+          deploymentClusters?.forEach((decisionCluster) => {
+            clusterWiseSubscriptionGroups[decisionCluster] =
+              clusterWiseSubscriptionGroups[decisionCluster] || [];
+            clusterWiseSubscriptionGroups[decisionCluster].push(
+              subscriptionGroup
+            );
+          });
+        });
```

Comment:
```suggestion
        subscriptionGroupInfo?.forEach((subscriptionGroup) => {
          const deploymentClusters: string[] =
            subscriptionGroup?.placement?.kind === ACMPlacementModel.kind
              ? getClustersFromDecisions(subscriptionGroup?.placementDecision)
              : getClustersFromDecisions(
                  subscriptionGroup?.placement as ACMPlacementRuleKind
                );

          deploymentClusters?.forEach((decisionCluster) => {
            clusterWiseSubscriptionGroups[decisionCluster] =
              clusterWiseSubscriptionGroups[decisionCluster] || [];
            clusterWiseSubscriptionGroups[decisionCluster].push(
              subscriptionGroup
            );
          });
        });
```
Please move this logic to a different function keep the function out of useSubscriptionParser, And declare this type and use it as return type.
```
type ClusterWiseSubscriptionGroupsMap: {
  [clusterName: string]: SubscriptionGroupType[];
}
```